flash/data/process.py

```diff
@@ -0,0 +1,188 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional, Sequence, Union
+
+import torch
+from pytorch_lightning.trainer.states import RunningStage
+from torch.nn import Module
+from torch.utils.data._utils.collate import default_collate
+
+from flash.data.batch import default_uncollate
+from flash.data.utils import convert_to_modules
+
+
+class Properties:
+
+    _running_stage = None
+
+    @property
+    def training(self) -> bool:
+        return self._running_stage == RunningStage.TRAINING
+
+    @training.setter
+    def training(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.TRAINING
+        elif self.training:
+            self._running_stage = None
+
+    @property
+    def testing(self) -> bool:
+        return self._running_stage == RunningStage.TESTING
+
+    @testing.setter
+    def testing(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.TESTING
+        elif self.testing:
+            self._running_stage = None
+
+    @property
+    def predicting(self) -> bool:
+        return self._running_stage == RunningStage.PREDICTING
+
+    @predicting.setter
+    def predicting(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.PREDICTING
+        elif self.predicting:
+            self._running_stage = None
+
+    @property
+    def validating(self) -> bool:
+        return self._running_stage == RunningStage.VALIDATING
+
+    @validating.setter
+    def validating(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.VALIDATING
+        elif self.validating:
+            self._running_stage = None
+
+
+class Preprocess(Properties, torch.nn.Module):
+
+    def __init__(
+        self,
+        train_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        valid_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        test_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        predict_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+    ):
+        super().__init__()
+        self.train_transform = convert_to_modules(train_transform)
+        self.valid_transform = convert_to_modules(valid_transform)
+        self.test_transform = convert_to_modules(test_transform)
+        self.predict_transform = convert_to_modules(predict_transform)
+
+    @classmethod
+    def load_data(cls, data: Any, dataset: Optional[Any] = None) -> Any:
```

Comment:
```suggestion
    def on_load_data(cls, data: Any, dataset: Optional[Any] = None) -> Any:
```
it does not load anything so the name is bit confusing

Reply:
It is for the base class. People should override it if needed.